R/fun_makefile_create.R

```diff
@@ -59,27 +60,29 @@ deepstate_create_makefile <-function(package,fun_name){
   dir.create(file.path(fun_path, paste0(fun_name,"_output")), showWarnings=FALSE, recursive=TRUE)
   file.create(makefile_path, recursive=TRUE)
 
-  obj.file.list <- Sys.glob(file.path(package, "src/*.so"))
-  if(length(obj.file.list) <= 0){
-    # try to generate the shared object before failing
+  shared_objects <- file.path(package, "src/*.so")  
+  if (length(Sys.glob(shared_objects)) <= 0) {
+    # ensure that the debugging symbols are embedded in the shared object
     makevars_file <- file.path(package, "src", "Makevars")
     if (dir.exists(file.path(package, "src"))) {
-      makevars_content <- "PKG_CXXFLAGS += -g \n"
-      write(makevars_content, makevars_file, append=TRUE)
+        makevars_content <- "PKG_CXXFLAGS += -g \n"
+        write(makevars_content, makevars_file, append=TRUE)
     }
 
-    system(paste0("R CMD INSTALL ", package), intern=FALSE)
-    
-    if (length(Sys.glob(file.path(package, "src/*.so"))) <= 0) {
-      stop("Missing package shared object file.")
+    install.packages(package, repo=NULL)
+
+    if (length(Sys.glob(shared_objects)) <= 0) {
+      error_msg <- paste("ERROR: the shared object for your package cannot be",
```

Comment:
you don't need a paste and intermediate `error_msg` variable here, instead just use
```r
stop(
  "the shared object ",
  "for your")
```
also please remove ERROR: since R adds that automatically, to avoid displaying Error: ERROR: etc.

Reply:
Fixed with commits f957075 and f739ebc